ports/stm32/boards/Passport/modules/flows/select_file_flow.py

```diff
@@ -0,0 +1,52 @@
+# SPDX-FileCopyrightText: 2023 Foundation Devices, Inc. <hello@foundationdevices.com>
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# select_file_flow.py - Decide what to do with a selected file
+
+from flows import Flow
+
+
+class SelectFileFlow(Flow):
+    def __init__(self, file_name, full_path, is_folder):
+        self.file_name = file_name
+        self.full_path = full_path
+        self.is_folder = is_folder
+        self.error = None
+        super().__init__(initial_state=self.choose_action, name='SelectFileFlow')
+
+    async def choose_action(self):
+        from pages import ChooserPage
+        options = [{'label': 'Navigate' if self.is_folder else 'Select', 'value': 0},
+                   {'label': 'Delete', 'value': 1}]
+
+        selection = await ChooserPage(options=options).show()
+
+        if selection is None:
+            self.set_result(None)
+            return
+
+        if selection == 1:
+            self.goto(self.delete_selected_file)
+        else:
+            self.set_result((self.file_name, self.full_path, self.is_folder))
+
+    async def delete_selected_file(self):
```

Comment:
Seems odd that the "select file flow" doesn't just select a file.  The choice of what to do with a selected file should probably be a separate piece of code.

Reply:
I changed the name of this to "SelectedFileFlow" for clarification, since this only decides what to do with a selected file.

Reply:
We should use the rightmost button for selecting stuff, and use the navpad right button to show stuff to do with the file, for example, deletion, maybe in the future we could add a "Details" option with file type (7z, PSBT, etc), size, etc.